src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3302,6 +3306,10 @@ def aks_agentpool_update(cmd,   # pylint: disable=unused-argument
         instance.max_count = None
 
     instance.tags = tags
+
+    if scale_down_mode is not None:
+        instance.scale_down_mode = scale_down_mode
```

Comment:
It seems that the default `scale_down_mode` for `nodepool add` is delete, and when the `nodepool update` operation is performed, if this option is not specified, the `scale_down_mode` will be set to None. Is this expected?

Reply:
how ill it be set `None` if we only update the `instance.scale_down` if it's not `None`?

Reply:
Oh my fault, I missed the if statement. The code changes look good, could you please add/update some test cases to cover this option?

Reply:
I had a validator before but turns out its not needed because the enum checks are built in the SDK. Are you asking for live tests for the field?

Reply:
Yes, I mean the live test.